src/colmap/estimators/bundle_adjustment.h

```diff
@@ -170,14 +172,22 @@ class BundleAdjuster {
 
   bool Solve(Reconstruction* reconstruction);
 
-  // Get the Ceres solver summary for the last call to `Solve`.
+  // Set up the problem
+  void SetUpProblem(Reconstruction* reconstruction,
```

Comment:
Would it make sense to pass the reconstruction to the constructor such that it doesn't need to be provided as argument most of its member functions? I can't see any use case where we need to use a different reconstruction between function calls. Maybe for future PRs @ahojnnes